sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuDeviceManager.scala

```diff
@@ -223,6 +224,11 @@ object GpuDeviceManager extends Logging {
     if (!Rmm.isInitialized) {
       val conf = rapidsConf.getOrElse(new RapidsConf(SparkEnv.get.conf))
       val info = Cuda.memGetInfo()
+
+      // We need to reserve more memory when RAPIDS shuffle is enabled and we are using the CUDA
+      // async allocator, so initialize the shuffle environment first.
+      GpuShuffleEnv.init(conf)
```

Comment:
but this is happening for all allocators right? I think this is probably where you want this for ASYNC, but not for the others?

Reply:
We need to always initialize it regardless of the allocator used, right?